Python/chapter03/p03_stack_of_plates/miguelHx.py

```diff
@@ -130,54 +130,69 @@ def __str__(self):
         return '->'.join(values)
 
 
+class SetOfStacksIterator(Generic[T]):
+    def __init__(self, stack_list: List[MyStack[T]]):
+        # build temporary list
+        self.my_list = []
+        self.index = 0
+        stack_list_reversed = stack_list[::-1]
+        for stack in stack_list_reversed:
+            for item in stack:
+                self.my_list.append(item)
```

Comment:
is there a way to make this iterator consume O(number_of_stacks) space instead of O(number_of_stacks * stack_threshold)? or even better, O(1) space!

Reply:
Yes, there is a way.  Used a generator, here are the changes: 9996dd81d91883dd50504bb6a721272432f09529
Space consumed by this iterator is now O(1) 😄